From 4e4550df7c6f69f93477a3da82f606f411d00f10 Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Thu, 17 Aug 2017 10:12:14 +0800
Subject: [PATCH] ARM: dts: rockchip: tidy up rk3229 related dt

Pick out non-public dt nodes from dtsi and separately configure in dts.

Change-Id: I774f45e0b282fb1b783b5b175fa4ffc9ad904990
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-echo-v10.dts | 12 ++++++++++++
 arch/arm/boot/dts/rk3229-evb.dts      | 28 +++++++++++++++++++++++++++
 arch/arm/boot/dts/rk3229-gva-sdk.dts  |  5 -----
 arch/arm/boot/dts/rk322x-android.dtsi | 26 -------------------------
 4 files changed, 40 insertions(+), 31 deletions(-)

diff --git a/arch/arm/boot/dts/rk3229-echo-v10.dts b/arch/arm/boot/dts/rk3229-echo-v10.dts
index fccddd6ab775..7c821d1696ab 100644
--- a/arch/arm/boot/dts/rk3229-echo-v10.dts
+++ b/arch/arm/boot/dts/rk3229-echo-v10.dts
@@ -311,6 +311,18 @@
 	assigned-clock-parents = <&xin24m>;
 };
 
+&sdmmc {
+	status = "okay";
+};
+
 &u2phy0_host {
 	phy-supply = <&vcc_host>;
 };
+
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/rk3229-evb.dts b/arch/arm/boot/dts/rk3229-evb.dts
index 027558b25771..637f4eb91e78 100644
--- a/arch/arm/boot/dts/rk3229-evb.dts
+++ b/arch/arm/boot/dts/rk3229-evb.dts
@@ -203,6 +203,10 @@
 	status = "okay";
 };
 
+&sdmmc {
+	status = "okay";
+};
+
 &u2phy0_host {
 	phy-supply = <&vcc_host>;
 };
@@ -210,3 +214,27 @@
 &u2phy1_host {
 	phy-supply = <&vcc_host>;
 };
+
+&usb_host0_ehci {
+	status = "okay";
+};
+
+&usb_host0_ohci {
+	status = "okay";
+};
+
+&usb_host1_ehci {
+	status = "okay";
+};
+
+&usb_host1_ohci {
+	status = "okay";
+};
+
+&usb_host2_ehci {
+	status = "okay";
+};
+
+&usb_host2_ohci {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/rk3229-gva-sdk.dts b/arch/arm/boot/dts/rk3229-gva-sdk.dts
index a70b617ef0ae..5a12414ed1f8 100644
--- a/arch/arm/boot/dts/rk3229-gva-sdk.dts
+++ b/arch/arm/boot/dts/rk3229-gva-sdk.dts
@@ -348,8 +348,3 @@
 	assigned-clocks = <&cru SCLK_SDIO_SRC>;
 	assigned-clock-parents = <&xin24m>;
 };
-
-&sdmmc {
-	status = "disabled";
-};
-
diff --git a/arch/arm/boot/dts/rk322x-android.dtsi b/arch/arm/boot/dts/rk322x-android.dtsi
index 8ba6e7987e87..f8c69f275da4 100644
--- a/arch/arm/boot/dts/rk322x-android.dtsi
+++ b/arch/arm/boot/dts/rk322x-android.dtsi
@@ -103,7 +103,6 @@
 };
 
 &sdmmc {
-	status = "okay";
 	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
@@ -115,7 +114,6 @@
 };
 
 &sdio {
-	status = "okay";
 	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
@@ -167,30 +165,6 @@
 	status = "okay";
 };
 
-&usb_host0_ehci {
-	status = "okay";
-};
-
-&usb_host0_ohci {
-	status = "okay";
-};
-
-&usb_host1_ehci {
-	status = "okay";
-};
-
-&usb_host1_ohci {
-	status = "okay";
-};
-
-&usb_host2_ehci {
-	status = "okay";
-};
-
-&usb_host2_ohci {
-	status = "okay";
-};
-
 &usb_otg {
 	status = "okay";
 };
-- 
2.35.3

